### ISSUE TITLE
Batch Import Resource Size Table Edits

### ISSUE THREAD

**Issue by AsteroidPizza39** [NX-Editor/NxEditor#43]
The purpose of this feature would be to increase the efficiency of editing the Resource Size Table for the sake of creating mods and increasing version compatibility. Here is how I would imagine this working, with the example file of `ResourceSizeTable.Product.112.rsizetable`. 

1. In nx-editor, open `ResourceSizeTable.Product.112.rsizetable`
2. In the Tools dropdown, there will be an option that reads `Batch Import RSTB Modifications`
3. Clicking that option prompts the user to select a txt file containing the paths and sizes of the modified files (An example of the content structure of this text file is attached)
4. The edits and additions to `ResourceSizeTable.Product.112.rsizetable` are applied automatically

![image](https://github.com/NX-Editor/NxEditor/assets/45804670/4c308b21-fe47-4712-a3b9-9f5ed22ed0dc)
[RSTB Edits.txt](https://github.com/NX-Editor/NxEditor/files/11854145/RSTB.Edits.txt)

**Comment by ArchLeaders** [comment]
One slight caveat, modifications/additions will need to be explicitly defined in case of duplicate hashes (which can happen fairly often with 300k values)

**Comment by AsteroidPizza39** [comment]
meaning we’d have to specify whether an edit is a modification or a new entry?

**Comment by ArchLeaders** [comment]
Correct.

**Comment by AsteroidPizza39** [comment]
Personally, i think that if someone is already creating a mod that edits RSTB, it’s a reasonable expectation that they’ll know which of their files is brand new vs which are edits.

**Comment by ArchLeaders** [comment]
Yes the user will know, but the computer needs to know as well.

**Comment by ArchLeaders** [comment]
So something like this is probably what I'd use for markup.

```json
{
    "StringTable": {
        "key_1": 1111,
        "key_2": 2222
    },
    "HashTable": {
        "key_3": 1111,
        "key_4": 2222
    }
}
```

Ideally, this will be generated after making edits in the restbl editor.

**Comment by AsteroidPizza39** [comment]
seeing as the program asks to define a game path for totk, you could add an ‘open TOTK restbl’ button with the TOTK plugin that would navigate to the restbl file of the dump. Then another button for importing/merging/exporting edits, and that would make nx-editor the ultimate mod merging tool (for now)

**Comment by ArchLeaders** [comment]
Yeah merging will be pretty trivial. I'll need some way for plugins to communicate to make it seamless though, which is a bit difficult.

**Comment by AsteroidPizza39** [comment]
if it would have to communicate between plugins, maybe base it in the main program and have it go through that?
